File: atividade-02/caixa.c
#include <stdio.h>
#include <stdbool.h>

int main(void)
{
    int   opcao;
    float saldo = 0.0f;
    float valor;
    bool  encerrar = false; 

    printf("=== CAIXA ELETRONICO ===\n");

    while (encerrar == false) {
        printf("\n--- MENU ---\n");
        printf("1 - Consultar saldo\n");
        printf("2 - Depositar\n");
        printf("3 - Sacar\n");
        printf("4 - Encerrar\n");
        printf("Escolha uma opcao: ");
        scanf("%d", &opcao);

        if (opcao == 1) {
            printf("Saldo atual: R$ %.2f\n", saldo);
        } else if (opcao == 2) {
            printf("Valor do deposito (R$): ");
            scanf("%f", &valor);

            if (valor > 0) {
                saldo = saldo + valor;
                printf("Deposito realizado. Novo saldo: R$ %.2f\n", saldo);
            } else {
                printf("Valor invalido. O deposito deve ser maior que zero.\n");
            }
        } else if (opcao == 3) {
            printf("Valor do saque (R$): ");
            scanf("%f", &valor);

            if (valor <= 0) {
                printf("Valor invalido. O saque deve ser maior que zero.\n");
            } else if (valor > saldo) {
                printf("Saldo insuficiente. Saldo atual: R$ %.2f\n", saldo);
            } else {
                saldo = saldo - valor;
                printf("Saque realizado. Novo saldo: R$ %.2f\n", saldo);
            }
        } else if (opcao == 4) {
            encerrar = true;
            printf("\nEncerrando... Saldo final: R$ %.2f\n", saldo);
            printf("Obrigado por utilizar nosso caixa eletronico!\n");
        } else {
            printf("Opcao inexistente. Tente novamente.\n");
        }
    }

    return 0;
}

File: atividade-04/caixa.py
# Atividade 04 - Caixa eletronico (atividade 02) refatorado para o paradigma funcional.

def mostrar_menu():
    print("\n--- MENU ---")
    print("1 - Consultar saldo")
    print("2 - Depositar")
    print("3 - Sacar")
    print("4 - Encerrar")


def depositar(saldo, valor):
    if valor > 0:
        print("Deposito realizado. Novo saldo: R$ %.2f" % (saldo + valor))
        return saldo + valor
    else:
        print("Valor invalido. O deposito deve ser maior que zero.")
        return saldo


def sacar(saldo, valor):
    if valor <= 0:
        print("Valor invalido. O saque deve ser maior que zero.")
        return saldo
    elif valor > saldo:
        print("Saldo insuficiente. Saldo atual: R$ %.2f" % saldo)
        return saldo
    else:
        print("Saque realizado. Novo saldo: R$ %.2f" % (saldo - valor))
        return saldo - valor


def executar(saldo):
    mostrar_menu()
    opcao = int(input("Escolha uma opcao: "))

    if opcao == 1:
        print("Saldo atual: R$ %.2f" % saldo)
        return executar(saldo)
    elif opcao == 2:
        valor = float(input("Valor do deposito (R$): "))
        return executar(depositar(saldo, valor))
    elif opcao == 3:
        valor = float(input("Valor do saque (R$): "))
        return executar(sacar(saldo, valor))
    elif opcao == 4:
        return saldo
    else:
        print("Opcao inexistente. Tente novamente.")
        return executar(saldo)


# --- programa principal ---

print("=== CAIXA ELETRONICO ===")

saldo_final = executar(0.0)

print("\nEncerrando... Saldo final: R$ %.2f" % saldo_final)
print("Obrigado por utilizar nosso caixa eletronico!")
